server/boot/user.js

```diff
@@ -56,6 +55,61 @@ function encodeFcc(value = '') {
   return replaceScriptTags(replaceFormAction(value));
 }
 
+function isAlgorithm(challenge) {
+  // test if name starts with hike/waypoint/basejump/zipline
+  // fix for bug that saved different challenges with incorrect
+  // challenge types
+  return !(/^(waypoint|hike|zipline|basejump)/i).test(challenge.name) &&
```

Comment:
I agree with @SaintPeter but this is too important to wait. Let's ship this as is and we can worry about doing this later.